theano/gof/fg.py

```diff
@@ -334,13 +334,15 @@ def find_path_to(output_var, input_var):
                             assert path is not None
                             tr = getattr(r.tag, 'trace', None)
                             detailed_err_msg = ""
-                            if tr:
-                                sio = StringIO()
-                                traceback.print_list(tr, sio)
-                                tr = sio.getvalue()
+                            if len(tr) > 0:
```

Comment:
`tr` can be `None` here, so its length is not defined in that case.
The easiest fix would be to change l. 335 to something like `tr = getattr(r.tag, 'trace', [])`.